### ISSUE TITLE
Syntax documentation

### ISSUE THREAD

**Issue by jhenin** [colvars/colvars#70]
The parser needs to be better documented, mostly with respect to braces and line breaks. I keep writing inputs that I expect to work but don't.

**Comment by giacomofiorin** [comment]
Post the examples as reply to this.  We certainly should fix any
inconsistency.

On Aug 9, 2016 12:54 PM, "Jérôme Hénin" <notifications@github.com> wrote:

The parser needs to be better documented, mostly with respect to braces and
line breaks. I keep writing inputs that I expect to work but don't.

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/colvars/colvars/issues/70>, or mute the thread

**Comment by jhenin** [comment]
For example, this is correct:
```
colvar {
name x
cartesian {
atoms { atomnumbers 1 2 3 }
}
}
```

but this isn't (and segfaults, btw):
```
colvar {
name x
cartesian { atoms { atomnumbers 1 2 3 } }
}
```

**Comment by giacomofiorin** [comment]
I didn't get the segfault, but no configuration for "atoms" shows up in the parser anyway.  Recompiling in debug mode, it gives:
```colvars:Initializing a new "cartesian" component, with configuration:```
```colvars: atoms { atomnumbers 1 2 3```
thus, the brace count is incorrect when isolating the config string for "cartesian".  It should be:
```colvars:Initializing a new "cartesian" component, with configuration:```
```colvars: atoms { atomnumbers 1 2 3 }```
I'll try to find the error in `key_lookup()` asap.

**Comment by giacomofiorin** [comment]
The above issue is fixed by 7e79388.  Leaving open for a little bit longer to see if any other parsing issues occur.

**Comment by jhenin** [comment]
Yes, that fixed what bothered me. Now in VMD I can finally define a colvar with a one-liners:
`cv config "colvar {name y \n cartesian { atoms { atomnumbers 1 2 } } }"`
